### PR TITLE
storage class assignment should not ignore errors

### PR DESCRIPTION
**Pull request by RomanBednar**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

Since having multiple default storage classes no longer produces errors (#110514), the caller (`assignDefaultStorageClass`) should no longer ignore errors - any error returned by `GetDefaultClass` is now a real error and assignment should fail.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 1]
```diff
@@ -955,10 +955,8 @@ func (ctrl *PersistentVolumeController) assignDefaultStorageClass(ctx context.Co
 
 	class, err := util.GetDefaultClass(ctrl.classLister)
 	if err != nil {
-		// It is safe to ignore errors here because it means we either could not list SCs or there is more than one default.
-		// TODO: do not ignore errors after this PR is merged: https://github.com/kubernetes/kubernetes/pull/110559
-		logger.V(4).Info("Failed to get default storage class", "err", err)
-		return false, nil
+		logger.Error(err, "Failed to get default storage class")
```
Comment (jsafrane) [inline pkg/controller/volume/persistentvolume/pv_controller.go:958]: The error will be logged in `updateClaim()`, no need to log it here.

File: pkg/volume/util/storageclass.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func GetDefaultClass(lister storagev1listers.StorageClassLister) (*storagev1.Sto
 		return defaultClasses[i].CreationTimestamp.UnixNano() > defaultClasses[j].CreationTimestamp.UnixNano()
 	})
 	if len(defaultClasses) > 1 {
-		klog.V(4).Infof("%d default StorageClasses were found, choosing the newest: %s", len(defaultClasses), defaultClasses[0].Name)
```
[the remark this answers is not in the document]
Comment (RomanBednar) [reply pkg/volume/util/storageclass.go:67]: That's just partially true, it uses the newest unless the timestamps are equal - that means this message could be misleading. This is what I'm referring to: https://github.com/kubernetes/kubernetes/blob/9e0569f2ed3934060fabe51be4e15232bbea3877/pkg/volume/util/storageclass.go#L62

File: pkg/controller/volume/persistentvolume/pv_controller.go (hunk 1 above)
Comment (RomanBednar) [reply pkg/controller/volume/persistentvolume/pv_controller.go]: Correct, removing.

**Pull request merged**